app/views/play/level/tome/Problem.coffee

```diff
@@ -104,24 +104,59 @@ module.exports = class Problem
       @ace.getSession().removeMarker @lineMarkerRange.id
       @lineMarkerRange.start.detach()
       @lineMarkerRange.end.detach()
+  
+  # Here we take a string from the locale file, find the placeholders ($1/$2/etc)
+  #   and replace them with capture groups (.+),
+  # returns a regex that will match against the error message
+  #   and capture any dynamic values in the text
+  makeTranslationRegex: (englishString) ->
+    escapeRegExp = (str) ->
+      # https://stackoverflow.com/questions/3446170/escape-string-for-use-in-javascript-regex
+      return str.replace(/[\-\[\]\/\{\}\(\)\*\+\?\.\\\^\$\|]/g, "\\$&")
+    new RegExp(escapeRegExp(englishString).replace(/\\\$\d/g, '(.+)').replace(/ +/g, ' +'))
 
   translate: (msg) ->
```

Comment:
Can you add some tests to this? I'm thinking a handful of errors in a few languages. In order to get around that none of these languages are translated, you could use rot13 or insert some fake (google?) translated errors into the locales during test prep.